govstack_api/building_blocks/bb_digital_registries/registries/registry_factory.py

```diff
@@ -1,16 +1,19 @@
 from govstack_api.building_blocks.bb_digital_registries.registries.insuree_registry import InsureeRegistry
+from django.apps import apps
 
 
 class RegistryFactory:
     def __init__(self):
-        self.registry_classes = {
-            ("registryname", "111"): InsureeRegistry,
-            # ("registryname2", "version2"): ProductRegistry,
-            # add more...
-        }
+        config = apps.get_app_config('govstack_api')
+        self.registry_config_data = config.registry_config_data
 
-    def get_registry_class(self, registryname: str, versionnumber: str, request):
-        registry_class = self.registry_classes.get((registryname, versionnumber))
-        if registry_class is None:
-            raise ValueError(f"Unsupported registry: {registryname} version: {versionnumber}")
-        return registry_class(request)
+    def get_registry(self, registry_name, version_number, request):
+        registry_config = self.registry_config_data.get(registry_name, {}).get(str(version_number))
+        if registry_config is not None:
```

Comment:
you could invert this is statement like: 
if registry_config is None: 
    raise ValueError(...) 
...
if registry_class is None: 
    raise ValueError(...) 
```
this way we shouldn't have a lot of nested if statements and while it doesn't reduce complexity of code, it's easier to read.

Reply:
Good idea, I will implement it